### PR TITLE
Setups

### PR DESCRIPTION
**Pull request by tawfiknasser**
#2 setting up the environment
Travis is missing, will be added later.
we setup the react and express.
the server package.json : 
`  "dependencies": {
    "cookie-parser": "~1.4.3",
    "express": "~4.16.0"
  },
  "devDependencies": {
    "debug": "~2.6.9",
    "eslint": "^5.5.0",
    "morgan": "~1.9.0",
    "nodemon": "^1.18.4"
  }
}
`

client package.json : 

`  "dependencies": {
    "react": "^16.5.0",
    "react-dom": "^16.5.0",
    "react-scripts": "1.1.5"
  },`

ready to work

### CODE REVIEW

**Comment by tawfiknasser** [comment]
should we gitignore package-lock ? #13
______

File: client/.gitignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/ignore-files/ for more about ignoring files.
```
Comment (shiryz) [inline client/.gitignore]: I don't see why you'd need more than one .gitignore in your files, the one in the root should suffice

File: app.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(express.static(path.join(__dirname, 'public')));
+
+app.use('/', indexRouter);
+
+
+// catch 404 and forward to error handler
+app.use(function(req, res, next) {
```
Comment (shiryz) [inline app.js]: Modularise in your router

File: app.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  next(createError(404));
+});
+
+// error handler
+app.use(function(err, req, res, next) {
```
Comment (shiryz) [inline app.js]: Modularise in router as well

File: app.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+var logger = require('morgan');
+
+var indexRouter = require('./routes/index');
+
+
+var app = express();
+
+// view engine setup
+app.set('views', path.join(__dirname, 'views'));
+app.set('view engine', 'jade');
```
Comment (shiryz) [inline app.js]: Why are you using both jade and react? 
jade is a templating engine and react is a frontend framework, eventually, they're doing the same job
use one

File: client/.gitignore (hunk 1 above)
Comment (tawfiknasser) [reply client/.gitignore]: b077f7bf082451873594bd4f524b1f29276e2a07

Comment (tawfiknasser) [reply client/.gitignore]: c5678d9b577e7cb17bbe95ab0cde55987510dd51

File: app.js (hunk 2 above)
Comment (tawfiknasser) [reply app.js]: 778caa06d586c8ad9dc20c5f436c722ef2626f79

**Comment by tawfiknasser** [comment]
@MynahMarie  
selection_018
here is our new files structure.
hope it's good now, thanks

**Comment by MynahMarie** [comment]
Also, your PR description could be edited to make it more clear on what you actually did to set up the environment. It should be more descriptive. For example, I like to write my PR description as follows:
```
In this PR, we have:
- set up the file system and divided it between client and server
- used Express-generator to create boiler plate files for the server side
- created a basic home route on the server
- set up the client side using Create-React-App

Ref: #2 
```

If you could do something similar that would greatly help the QA's job to review your code. It makes it clear and simple to track your changes and understand what you are trying to do.

File: client/src/App.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import "./App.css";
+
+const App = () => {
+  return (
+    <div className="App">
+      <header>
+        <h1>Election-Day App</h1>
+      </header>
+
+      <h3>Hello World</h3>
```
Comment (shiryz) [inline client/src/App.js]: Unnecessary

Comment (shiryz) [inline client/src/App.js]: unnecessary, please remove

File: client/src/index.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
```
Comment (shiryz) [inline client/src/index.js]: why is this imported here while never used?

File: client/src/App.css [hunk 7]
```diff
@@ -0,0 +1,3 @@
+.App {
```
Comment (shiryz) [inline client/src/App.css]: Give it a better name please

File: client/package.json [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    "react-dom": "^16.5.0",
+    "react-scripts": "1.1.5"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test --env=jsdom",
+    "eject": "react-scripts eject"
+  },
+  "proxy": "http://localhost:3000"
```
Comment (MynahMarie) [inline client/package.json]: I'm not sure this is correct. Here is an example of how I used this in previous projects:

```
"proxy": {
    "/*": {
      "target": "http://localhost:3000"
     }
  }

File: client/public/manifest.json [hunk 9]
```diff
@@ -0,0 +1,15 @@
+{
+  "short_name": "React App",
+  "name": "Create React App Sample",
```
Comment (MynahMarie) [inline client/public/manifest.json]: Here as well, the name should be changed.

File: package.json [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+  "scripts": {
+    "start": "nodemon ./server/index"
+  },
+  "dependencies": {
+    "cookie-parser": "~1.4.3",
+    "express": "~4.16.0"
+  },
+  "devDependencies": {
+    "debug": "~2.6.9",
+    "eslint": "^5.5.0",
```
Comment (MynahMarie) [inline package.json]: Great that you added eslint but where is the configuration file (`.eslintrc`). This file should be in the root folder of your project and should contained some configuration for basic rules your team agrees with.

File: server/index.js [hunk 11]
```diff
@@ -0,0 +1,5 @@
+const app = require("./app");
+const port = process.env.PORT || "3000";
+app.set("port", port);
```
Comment (MynahMarie) [inline server/index.js]: Just a minor change here, for the purpose of following conventions its better to write environment variables in all capital letters. So `port` should be `PORT`. This communicates where this variable lives more clearly.

File: client/package.json (hunk 8 above)
Comment (tawfiknasser) [reply client/package.json]: we tried it, and it's working.
if any changes happens i will use yours. thanks

**Comment by shiryz** [comment]
Hallelujah :pray:

**Pull request merged**